src/Sector.cpp

```diff
@@ -76,23 +88,40 @@ QPair<double, double> Sector::getCenter() const {
     runningTotal.first = 0;
     runningTotal.second = 0;
     const int count = points.size();
-    for(int i = 0; i < points.size(); ++i) {
-        A += (points[i].first * points[(i + 1) % count].second
-             - points[(i + 1) % count].first * points[i].second);
 
-        double multiplyBy = points[i].first * points[(i + 1) % count].second
-                            - (points[(i + 1) % count].first * points[i].second);
+    if(count == 0) {
+        qDebug() << "Sector::getCenter() Sector " << name << "(" << icao << ") doesn't contain any points";
```

Comment:
This would currently return `QPair((double) 0.0, (double) 0.0)` on error. Consuming code would have to check if that is the case (with fuzziness because floating point...). This is not sane IMHO.

We could refactor the function to be able to signal "no valid center" to consuming code, e.g. with a nullptr. But I rather think this code should never be called on an empty list.

I would just die here.

Something like this:
```
qCritical() << "Sector::getCenter() Sector " << name << "(" << icao << ") doesn't contain any points";
QTextStream(stdout) << "CRITICAL: Sector " << name << "(" << icao << ") doesn't contain any points" << '\n';
exit(EXIT_FAILURE);
```

Reply:
(The `qCritical()` makes sure we have a proper log.txt entry)